modules/tensor_mechanics/src/materials/ComputeLinearViscoelasticStress.C

```diff
@@ -0,0 +1,35 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeLinearViscoelasticStress.h"
+
+template <>
+InputParameters
+validParams<ComputeLinearViscoelasticStress>()
+{
+  InputParameters params = validParams<ComputeLinearElasticStress>();
+  params.addClassDescription("Divides total strain into elastic + creep + eigenstrains");
+  params.addParam<std::string>("creep_strain_name", "creep_strain", "name of the creep strain");
+  return params;
+}
+
+ComputeLinearViscoelasticStress::ComputeLinearViscoelasticStress(const InputParameters & parameters)
+  : ComputeLinearElasticStress(parameters),
```

Comment:
Just something to think about - i wonder whether inheriting from StressUpdateBase would be better?  I'm not sure whether this stuff will be used in conjunction with plasticity models.  What do you think, @bwspenc ?